Description: 修复 x86 自动打包环境下系统 vendor 检测错误问题
 当前，x86 自动打包环境（shuttle）中，/etc/dpkg/origins/default
 实际指向 debian 对应的文件。这会造成系统 vendor 检测错误，
 使得对应的 keyring 包解析错误。本次提交通过提前检测 /etc/os-release 
 临时修复了该问题。等待后续 x86 自动打包环境完成修复后，
 可以安全移除本次提交。
Author: miaochangwei <miaochangwei@uniontech.com>
Origin: https://gerrit.uniontech.com/plugins/gitiles/base/apt/+/40cf7ceb5eed8e57ede150301968383aa11d798b
Last-Update: 2022-05-25

--- apt-2.4.5.orig/vendor/getinfo
+++ apt-2.4.5/vendor/getinfo
@@ -4,6 +4,19 @@
 BASEDIR="$(readlink -f "$(dirname $0)")"
 
 getcurrent() {
+	# for some reason, in uos/deepin's automatic package system,
+	# /etc/dpkg/origins/default may link to /etc/dpkg/origins/debian.
+	# This should be removed in the future.
+	if [ -f /etc/dpkg/origins/deepin ]; then
+		if grep -isq 'UOS' /etc/os-release > /dev/null; then
+			echo uos
+			return 0
+		fi
+		if grep -isq 'Deepin' /etc/os-release > /dev/null; then
+			echo deepin
+			return 0
+		fi
+	fi
 	# search for an exact match to use the correct sources.list example
 	cd $BASEDIR
 	DISTROS="$(find . -mindepth 1 -maxdepth 1 -type d | cut -d'/' -f 2 | LC_ALL=C sort)"
